Wikipedia/Code/WMFArticleViewController.m

```diff
@@ -531,69 +531,68 @@ - (void)updateToolbarItemEnabledState {
+        
+         [_hideTableOfContentsToolbarItem applyTheme:self.theme];
     }
     return _hideTableOfContentsToolbarItem;
 }
 
-- (UIBarButtonItem *)saveToolbarItem {
+- (IconBarButtonItem *)saveToolbarItem {
     if (!_saveToolbarItem) {
-        UIButton *button = [UIButton buttonWithType:UIButtonTypeCustom];
-        button.frame = CGRectMake(0, 0, 25, 25);
-        [button setImage:[UIImage imageNamed:@"save"] forState:UIControlStateNormal];
-        [button addTarget:self action:@selector(toggleSave:event:) forControlEvents:UIControlEventTouchUpInside];
+        _saveToolbarItem = [[IconBarButtonItem alloc] initWithIconName: @"save" target: self action:@selector(toggleSave:event:) for: UIControlEventTouchUpInside];
         UILongPressGestureRecognizer *longPress = [[UILongPressGestureRecognizer alloc] initWithTarget:self action:@selector(handleSaveButtonLongPressGestureRecognizer:)];
-        [button addGestureRecognizer:longPress];
-        _saveToolbarItem = [[UIBarButtonItem alloc] initWithCustomView:button];
+        if ([_saveToolbarItem.customView isKindOfClass:[UIButton class]]) {
+            UIButton *button = (UIButton *)_saveToolbarItem.customView;
+             [button addGestureRecognizer:longPress];
+        }
+        [_saveToolbarItem applyTheme:self.theme];
     }
     return _saveToolbarItem;
 }
 
-- (UIBarButtonItem *)readingThemesControlsToolbarItem {
+- (IconBarButtonItem *)readingThemesControlsToolbarItem {
     if (!_readingThemesControlsToolbarItem) {
-        _readingThemesControlsToolbarItem = [[UIBarButtonItem alloc] initWithImage:[UIImage imageNamed:@"font-size"] style:UIBarButtonItemStylePlain target:self action:@selector(showReadingThemesControlsPopup)];
+        _readingThemesControlsToolbarItem = [[IconBarButtonItem alloc] initWithIconName: @"font-size" target: self action:@selector(showReadingThemesControlsPopup) for: UIControlEventTouchUpInside];
+        [_readingThemesControlsToolbarItem applyTheme:self.theme];
     }
     _readingThemesControlsToolbarItem.accessibilityLabel = WMFLocalizedStringWithDefaultValue(@"article-toolbar-reading-themes-controls-toolbar-item", nil, nil, @"Reading Themes Controls", @"Accessibility label for the Reading Themes Controls article toolbar item");
     return _readingThemesControlsToolbarItem;
 }
 
-- (UIBarButtonItem *)shareToolbarItem {
+- (IconBarButtonItem *)shareToolbarItem {
     if (!_shareToolbarItem) {
-        _shareToolbarItem = [[UIBarButtonItem alloc] initWithBarButtonSystemItem:UIBarButtonSystemItemAction
-                                                                          target:self
-                                                                          action:@selector(shareArticle)];
+        _shareToolbarItem = [[IconBarButtonItem alloc] initWithIconName: @"share" target: self action:@selector(shareArticle) for: UIControlEventTouchUpInside];
+        _shareToolbarItem.accessibilityLabel = WMFCommonStrings.accessibilityShareTitle;
```

Comment:
Good catch, thanks for fixing the missing accessibility label!

Reply:
@joewalsh thanks but it was still accessible before for me, I guess because it was a system button :)